src/figuresClasses.ts

```diff
@@ -1,11 +1,82 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Sides must be greater than 0');
+    }
+
+    const sides = [a, b, c].sort((x, y) => x - y);
 
-export class Triangle implements Figure {}
+    if (sides[2] >= sides[0] + sides[1]) {
+      throw new Error(
+        'The longest side must be greater than sum of two others',
+      );
+    }
+  }
+
+  getArea(): number {
+    const halfPerimetr = (this.a + this.b + this.c) / 2;
+
+    return (
+      Math.floor(
+        Math.sqrt(
+          halfPerimetr *
+            (halfPerimetr - this.a) *
+            (halfPerimetr - this.b) *
+            (halfPerimetr - this.c),
+        ) * 100,
+      ) / 100
+    );
+  }
+}
 
-export class Circle implements Figure {}
+export class Circle implements Figure {
+  shape: string = 'circle';
 
-export class Rectangle implements Figure {}
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public radius: number,
+  ) {
+    if (radius <= 0) {
```

Comment:
The variable 'radius' should be prefixed with 'this.' to refer to the class property. It should be 'this.radius <= 0'.